File: mobile/lib/data/models/facility.dart
class Facility {
  const Facility({
    required this.id,
    required this.buildingId,
    required this.name,
    required this.category,
  });

  final String id;
  final String buildingId;
  final String name;
  final String category;

  factory Facility.fromJson(Map<String, dynamic> json) => Facility(
        id: json['id'] as String,
        buildingId: json['buildingId'] as String,
        name: json['name'] as String,
        category: json['category'] as String? ?? 'general',
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'buildingId': buildingId,
        'name': name,
        'category': category,
      };
}

File: backend/app/models/facility.py
from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column

from app.database.session import Base


class Facility(Base):
    __tablename__ = "facilities"

    id: Mapped[str] = mapped_column(String, primary_key=True)
    building_id: Mapped[str] = mapped_column(String, index=True, nullable=False)
    name: Mapped[str] = mapped_column(String, nullable=False)
    category: Mapped[str] = mapped_column(String, default="general")
